dev_apps/music-nga/views/home/view.js

```diff
@@ -21,60 +45,225 @@ var HomeView = View.extend(function HomeView() {
 
   this.client.on('databaseChange', () => this.update());
 
+  this.renderPlaceholders(10);
   this.update();
+
+  debug('initialized');
 });
 
+/**
+ * Gets fresh data and re-renders
+ * the entire list.
+ *
+ * @private
+ */
 HomeView.prototype.update = function() {
   this.getAlbums().then((albums) => {
+    debug('got albums', albums);
     this.albums = albums;
     this.render();
   });
 };
 
-HomeView.prototype.destroy = function() {
-  this.client.destroy();
-
-  View.prototype.destroy.call(this); // super(); // Always call *last*
-};
-
+/**
+ * The title displayed in the
+ * main app chrome header.
+ *
+ * @type {String}
+ */
 HomeView.prototype.title = 'Music';
 
+/**
+ * Renders a tile for each album.
+ *
+ * If the isn't a spare tiel for the album
+ * a new one will be made. Empty tiles
+ * will be cleaned up at the end.
+ *
+ * @private
+ */
 HomeView.prototype.render = function() {
   View.prototype.render.call(this); // super();
 
-  var html = '';
+  this.albums.forEach((album, i) => {
+    var tile = this.tiles[i] || this.createTile();
+    var root = tile.root;
 
-  this.albums.forEach((album) => {
-    var template =
-`<a class="tile"
-    href="/player?id=${album.name}"
-    data-artist="${album.metadata.artist}"
-    data-album="${album.metadata.album}"
-    data-file-path="${album.name}">
-  <img>
-</a>`;
+    root.href = `/player?id=${album.name}`;
+    tile.artist.textContent = `${album.metadata.artist}`;
+    tile.album.textContent = `${album.metadata.album}`;
+    root.dataset.filePath = `${album.name}`;
+    root.classList.add('populated');
 
-    html += template;
+    if (!root.parentNode) this.els.tiles.appendChild(root);
   });
 
-  this.tiles.innerHTML = html;
+  this.cleanEmptyTiles();
+  this.renderImages();
+};
+
+/**
+ * Because we recycle the tiles, there can
+ * sometimes be more tiles than albums.
+ *
+ * This method removes redundant tiles
+ * from the DOM and this.tiles list.
+ *
+ * @private
+ */
+HomeView.prototype.cleanEmptyTiles = function() {
+  debug('clean empty tiles');
+  var end = this.albums.length;
+  var empty = this.tiles.splice(end);
+  empty.forEach(tile => tile.root.remove());
+};
+
+/**
+ * Renders all tile image in batches lazily.
+ *
+ * @private
+ */
+HomeView.prototype.renderImages = function() {
+  debug('render images');
+
+  var total = this.tiles.length - 1;
+  var defer = deferred();
```

Comment:
This is a pattern I picked up from SMS team. I wasn't a big fan at first but started to like it as you don't always want to resolve the promise inside the callback it was created in. It also saves you from getting into a 'pyramid of doom' :)

Having said that, I'm happy to change to the `new Promise` pattern for consistency :smile: